src/data/value/into.rs

```diff
@@ -0,0 +1,91 @@
+use {
+    super::{Value, ValueError},
+    crate::result::Result,
+};
+
+impl Into<String> for Value {
+    fn into(self) -> String {
+        match self {
+            Value::Str(value) => value,
+            Value::Bool(value) => (if value { "TRUE" } else { "FALSE" }).to_string(),
+            Value::I64(value) => value.to_string(),
+            Value::F64(value) => value.to_string(),
+            Value::Null => String::from("NULL"),
+        }
+    }
+}
+
+pub trait TryInto<T>: Sized {
+    fn try_into(&self) -> Result<T>;
+}
+
+impl TryInto<bool> for Value {
+    fn try_into(&self) -> Result<bool> {
+        Ok(match self {
+            Value::Bool(value) => *value,
+            Value::I64(value) => match value {
+                1 => true,
+                0 => false,
+                _ => return Err(ValueError::ImpossibleCast.into()),
+            },
+            Value::F64(value) => {
+                if value.eq(&1.0) {
+                    true
+                } else if value.eq(&0.0) {
+                    false
+                } else {
+                    return Err(ValueError::ImpossibleCast.into());
+                }
+            }
+            Value::Str(value) => match value.to_uppercase().as_str() {
+                "TRUE" => true,
+                "FALSE" => false,
+                _ => return Err(ValueError::ImpossibleCast.into()),
+            },
+            Value::Null => return Err(ValueError::ImpossibleCast.into()),
+        })
+    }
+}
+impl TryInto<i64> for Value {
+    fn try_into(&self) -> Result<i64> {
+        Ok(match self {
+            Value::Bool(value) => {
+                if *value {
+                    1
+                } else {
+                    0
+                }
+            }
+            Value::I64(value) => *value,
+            Value::F64(value) => value.trunc() as i64,
+            Value::Str(value) => value
+                .parse::<i64>()
+                .map_err(|_| ValueError::ImpossibleCast)?,
+            Value::Null => return Err(ValueError::ImpossibleCast.into()),
+        })
+    }
+}
+impl TryInto<f64> for Value {
+    fn try_into(&self) -> Result<f64> {
+        Ok(match self {
+            Value::Bool(value) => {
+                if *value {
+                    1.0
+                } else {
+                    0.0
+                }
+            }
+            Value::I64(value) => (*value as f64).trunc(),
+            Value::F64(value) => *value,
+            Value::Str(value) => value
+                .parse::<f64>()
+                .map_err(|_| ValueError::ImpossibleCast)?,
+            Value::Null => return Err(ValueError::ImpossibleCast.into()),
+        })
+    }
+}
+impl TryInto<String> for Value {
+    fn try_into(&self) -> Result<String> {
+        Ok(self.to_owned().into())
+    }
+}
```

Comment:
```rust
impl Into<String> for Value {
    fn into(self) -> String { .. }
}
```
I think this is enough which you already added well.

`Value` -> `String` does not make any `Error`.
Let's simply use `Into`, without `TryInto<String>`

Reply:
My thinking was that everything else has `try_into()` but string doesn't. I figured it might be a bit confusing if it is missing.

Any ideas on how to solve that? Or is it reasonable to let it be missing?

Reply:
Oh, sorry I missed the comment, I think it's reasonable not to have TryInto.
I think we can simply consider this kind of work in systemic way first.

By adding `TryInto<String> for Value` can make code might look slightly better but it causes the system confusing because `Value -> String` should not cause error but `TryInto<String> for Value` implies that failure is possible.

> My thinking was that everything else has try_into() but string doesn't.

Simple to think about this, because everything else can fail but string cannot.

Reply:
Yeah. I suppose it's not like you will write code that sometimes gets a String, sometimes gets an i64. This doesn't prevent loops from being able to happen.

Though it may have some macro implications. I see your point and it is fair enough.